code/modules/power/cable.dm

```diff
@@ -549,32 +586,173 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list(new/datum/stack_recipe("cable restrain
+	vis_contents += cable_node_2
+	cable_node_3 = new /obj/effect/node/layer3()
+	vis_contents += cable_node_3
+	update_icon()
+
+/obj/structure/cable/multilayer/Destroy()					// called when a cable is deleted
+	QDEL_NULL(machinery_node)
+	QDEL_NULL(cable_node_1)
+	QDEL_NULL(cable_node_2)
+	QDEL_NULL(cable_node_3) 
+	return ..()									// then go ahead and delete the cable
+
+/obj/structure/cable/multilayer/examine(mob/user)
+	. += ..()
+	. += "<span class='notice'>L1:[cable_layer & CABLE_LAYER_1 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>L2:[cable_layer & CABLE_LAYER_2 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>L3:[cable_layer & CABLE_LAYER_3 ? "Connect" : "Disconnect"].</span>"
+	. += "<span class='notice'>M:[machinery_layer & MACHINERY_LAYER_1 ? "Connect" : "Disconnect"].</span>"
+
+/obj/structure/cable/multilayer/attack_hand(mob/living/user)
+	if(!user)
+		return
+	var/list/layer_list = list(
+		"Layer 1" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-red"),
+		"Layer 2" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-yellow"),
+		"Layer 3" = image(icon = 'icons/mob/radial.dmi', icon_state = "coil-blue"),
+		"Machinery" = image(icon = 'icons/obj/power.dmi', icon_state = "smes")
+		)
+	var/layer_result = show_radial_menu(user, src, layer_list, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
+		return
+	var/CL
+	switch(layer_result)
+		if("Layer 1")
+			CL = CABLE_LAYER_1
+			to_chat(user, "<span class='warning'>You toggle L1 connection.</span>")
+		if("Layer 2")
+			CL = CABLE_LAYER_2
+			to_chat(user, "<span class='warning'>You toggle L2 connection.</span>")
+		if("Layer 3")
+			CL = CABLE_LAYER_3
+			to_chat(user, "<span class='warning'>You toggle L3 connection.</span>")
+		if("Machinery")
+			if(machinery_layer & MACHINERY_LAYER_1)
+				machinery_layer ^= MACHINERY_LAYER_1
+			else
+				machinery_layer |= MACHINERY_LAYER_1
+			to_chat(user, "<span class='warning'>You toggle machinery connection.</span>")
+
+	cut_cable_from_powernet(FALSE)
+
+	Disconnect_cable()
+
+	if(cable_layer & CL)
+		cable_layer ^= CL
+	else
+		cable_layer |= CL
+
+	Connect_cable(TRUE)
+
+	Reload()
+
+/obj/structure/cable/multilayer/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
 	return TRUE
 
-#undef UNDER_SMES
-#undef UNDER_TERMINAL
+///Reset powernet in this hub.
+/obj/structure/cable/multilayer/proc/Reload()
+	var/turf/T = get_turf(src)
+	for(var/obj/structure/cable/C in T.contents - src)
+		if(C.cable_layer & cable_layer)
+			C.deconstruct()						// remove adversary cable
+	auto_propagate_cut_cable(src)				// update the powernets
+
+/obj/structure/cable/multilayer/CtrlClick(mob/living/user)
+	to_chat(user, "<span class='warning'>You pust reset button.</span>")
+	addtimer(CALLBACK(src, .proc/Reload), 10, TIMER_UNIQUE) //spam protect

```

Comment:
```suggestion
	auto_propagate_cut_cable(src)				// update the powernets
	reloading = FALSE

/obj/structure/cable/multilayer/CtrlClick(mob/living/user)
	if(reloading)
		return
	reloading = TRUE
	to_chat(user, "<span class='warning'>You pust reset button.</span>")
	addtimer(CALLBACK(src, .proc/Reload), 1 SECONDS, TIMER_UNIQUE) //spam protect
```
A little spam protection?

Reply:
TIMER_UNIQUE dont allow to create new timer. When you click again before last timer end it work nothing happens.

Reply:
This is way more expensive than a var-check (besides the callback it will create and destroy a timed event, after generating and comparing the hash), but it's also prettier code.

Up to you, I'd go with the var-check instead, but it's not that important as this is not used often.